src/loops/simple_programs_with_repitition/easy/q8/MainTest.java

```diff
@@ -0,0 +1,58 @@
+package loops.simple_programs_with_repitition.easy.q8;
+
+import global.variables.*;
+import global.variables.clauses.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "This is a sentence. This is another sentence." + System.lineSeparator() + "2";
+
+        return new Clause[] {
+            new StringLiteral("Enter a sentence: "),
+            new NewLine(),
+            new StringLiteral("Enter a number to offset by: "),
+            new NewLine(),
+            new StringLiteral("Your Encrypted Sentence is: "),
+            new PlaceHolder()
+        };
+        
+    }
+
+    public void runMain() {
+        CaesarSike.main(new String[0]);
+    }
+
+    public static String encrypt(String sentence, int offset) {
+        String result = "";
+        int i = 0;
+        while (i < sentence.length()) {
+            result += (char)(sentence.charAt(i) + offset);
+            i++;
+        }
+        return result;
+    }
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of("This is just for testing. Do not copy!", 1), Arguments.of("This is second test", 100), Arguments.of(" ", 0)); // Doesn't work with empty string
```

Comment:
What about negative offsets? Also the same comment about empty strings not working - that is an important thing to raise when you see it as either the question or the test needs to change.

I'm adding the bad question tag for this reason